src/process.c

```diff
@@ -778,6 +778,246 @@ cc_free_pointer(gpointer str)
+		}
+		goto out;
+	}
+
+	/*
+	 * parse the sym_link in order to grab the bdf and make sure it is a
+	 * pci device.  If it is a virtual device, skip out.
+	 */
+	tokens = g_strsplit (sym_link, "/", 8);
+	if (!tokens) {
+		g_debug("sym_link read failure!");
+		goto out;
+	}
+
+	if ( !(g_strcmp0(tokens[3], "virtual")) ) {
+		g_debug("virtual device found");
+		goto out;
+	}
+
+	/* Grab BDF */
+	device_dir_path = g_strdup_printf("/sys/class/net/%s/device", sym_link);
+	sym_link = g_file_read_link(device_dir_path, &link_error);
+	if (!sym_link) {
+		if (link_error) {
+			g_debug ("device relative-path sym_link read failure. Path: %s", tmp_path);
+		}
+		goto out;
+	}
+
+	tokens = g_strsplit (sym_link, "/", 4);
+	if (!tokens || !tokens[3] ) {
+		g_debug("unexpected device path format\n");
+		goto out;
+	}
+
+	/* Grab device ID */
+	tmp_path = g_strdup_printf("%s/device", device_dir_path);
+	if ( !(g_file_get_contents(tmp_path, &buffer, NULL, NULL)) ) {
+		g_debug("Reading file %s returned error", tmp_path);
+		goto out;
+	}
+	g_free_if_set(tmp_path);
+	device_id_from_file = g_strsplit (buffer, "\n", -1);
+	if (!device_id_from_file) {
+		g_debug ("unexpected file format for address. buffer: %s\n", buffer);
+		goto out;
+	}
+	g_free(buffer);
+	
+	/* Grab vendor ID */
+	tmp_path = g_strdup_printf("%s/vendor", device_dir_path);
+	if ( !(g_file_get_contents(tmp_path, &buffer, NULL, NULL)) ) {
+		g_debug("Reading file %s returned error", tmp_path);
+		goto out;
+	}
+	g_free_if_set(tmp_path);
+	vendor_id_from_file = g_strsplit (buffer, "\n", -1);
+	if (!vendor_id_from_file) {
+		g_debug ("unexpected file format for address. buffer: %s\n", buffer);
+		goto out;
+	}
+	g_free(buffer);
+
+	vendor_device_id = g_strdup_printf("%s %s", vendor_id_from_file[0], device_id_from_file[0]);
+	g_free(device_id_from_file);
+	g_free(vendor_id_from_file);
+
+	/* Grab mac address */
+	tmp_path =  g_strdup_printf("/sys/class/net/%s/address", dirname);
+	if ( !(g_file_get_contents(tmp_path, &buffer, NULL, NULL)) ) {
+		g_debug("Reading address file %s returned error", tmp_path);
+		goto out;
+	}
+	g_free_if_set(tmp_path);
+
+	mac_from_file = g_strsplit (buffer, "\n", -1);
+	if (!mac_from_file) {
+		g_debug ("unexpected file format for address. buffer: %s\n", buffer);
+		goto out;
+	}
+
+	/*
+	 * get driver-path.  Symlink of /sys/bus/pci/devices/_/driver
+	 * should resolve to ../../../../bus/pci/drivers/<driver>
+	 */
+	tmp_path = g_strdup_printf("/sys/bus/pci/devices/%s/driver", tokens[3]);
+	sym_link = g_file_read_link(tmp_path, &link_error);
+
+	if (!sym_link || link_error) {
+		g_debug("driver-path sym_link read failure");
+		goto out;
+	}
+	g_free_if_set(tmp_path);
+
+	driver_tokens = g_strsplit(sym_link, "/", 8);
+
+	/* pending the device, the location of driver token will vary, but
+	 * will always be the last token
+	 */
+
+	if (!driver_tokens) {
+		g_debug ("unexpected driver-path format for %s", sym_link);
+		goto out;
+	}
+	while( driver_tokens[idx] )
```

Comment:
If you know there are 8 fields, it might be better to use a `for` loop to avoid any possibility of an infinite loop here.